### DISCUSSION TITLE
How to test the components that uses vuex with injection key

### DISCUSSION THREAD

**Discussion by tomocrafter** [vuejs/test-utils#588] [Q&A] [answered]
since `useStore` function of vuex accpets injection key for typing the store, but vue-test-utils-next only accepts plugins.
Is there any way to specify injection key for store?

(If I call `useStore` in component's setup on test, it will return undefined because injection key is not registered correctly. it works if not in test though)

**Comment by tomocrafter** [comment, marked as answer]
I completely overlooked that https://next.vue-test-utils.vuejs.org/api/#global plugins options accepts arguments if passed it with array.

**Comment by cexbrayat** [comment]
I usually use `provide` when I mount the component to register a fake store (or the real one) with the correct key:

```ts
const wrapper = mount(AppWithVuex, { global: { provide: { [myStoreKey]: store } } });
```
